tidy3d/plugins/mode/solver.py

```
@@ -212,6 +216,26 @@ def compute_modes(
         else:
             target_neff_p *= 1 + TARGET_SHIFT
 
+        # preprocess solver_basis_fields
+        basis_E = None
+        if solver_basis_fields is not None:
+            basis_E = solver_basis_fields[:3, ...]
+            try:
+                basis_E = basis_E.reshape((3, Nx * Ny, num_modes))
+            except ValueError as e:
+                log.error(
```

let's raise an exception (maybe another ValueError) instead of `log.error()`.